aptos-move/aptos-vm/src/sharded_block_executor/sharded_executor_client.rs

```diff
@@ -67,19 +75,25 @@ impl ShardedExecutorClient {
         let mut cross_shard_msg_txs = vec![];
         let mut cross_shard_msg_rxs = vec![];
         for _ in 0..num_shards {
-            let (messages_tx, messages_rx) = std::sync::mpsc::channel();
-            cross_shard_msg_txs.push(messages_tx);
-            cross_shard_msg_rxs.push(messages_rx);
+            let mut current_shard_msg_txs = vec![];
```

Comment:
probably can do this part with bunch of chained iterators and collects, but whatever we prefer.

Reply:
Let me take this as a follow-up to see if we can do it more rust style :)